vault/programs/vault/src/context.rs

```diff
@@ -0,0 +1,323 @@
+        seeds = [user_authority.key().as_ref(),
+            vault.vault_name.as_ref().strip(),
+            b"user_redeemable"],
+        bump,
+        payer = user_authority)]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+    // vault Accounts
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault)]
+    pub vault: Box<Account<'info, Vault>>,
+    #[account(seeds = [vault.vault_name.as_ref().strip(), b"payer"],
+        bump = vault.bumps.vault_payer)]
+    pub vault_payer: AccountInfo<'info>,
+    #[account(seeds = [vault.vault_name.as_ref().strip(), b"redeemable_mint"],
+        bump = vault.bumps.redeemable_mint)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+    // Programs and Sysvars
+    pub system_program: Program<'info, System>,
+    pub token_program: Program<'info, Token>,
+    pub rent: Sysvar<'info, Rent>,
+}
+
+#[derive(Accounts)]
+pub struct ExchangeUsdcForRedeemable<'info> {
+    // User Accounts
+    pub user_authority: Signer<'info>,
+    // TODO replace these with the ATA constraints when possible
+    #[account(mut,
+        constraint = user_usdc.owner == user_authority.key(),
+        constraint = user_usdc.mint == usdc_mint.key())]
+    pub user_usdc: Box<Account<'info, TokenAccount>>,
+    #[account(mut,
+        seeds = [user_authority.key().as_ref(),
+            vault.vault_name.as_ref().strip(),
+            b"user_redeemable"],
+        bump)]
+    pub user_redeemable: Box<Account<'info, TokenAccount>>,
+    // vault Accounts
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = usdc_mint)]
+    pub vault: Box<Account<'info, Vault>>,
+    #[account(seeds = [vault.vault_name.as_ref().strip(), b"payer"],
+        bump = vault.bumps.vault_payer)]
+    pub vault_payer: AccountInfo<'info>,
+    pub usdc_mint: Box<Account<'info, Mint>>,
+    #[account(mut,
+        seeds = [vault.vault_name.as_ref().strip(), b"redeemable_mint"],
+        bump = vault.bumps.redeemable_mint)]
+    pub redeemable_mint: Box<Account<'info, Mint>>,
+    #[account(mut,
+        seeds = [vault.vault_name.as_ref().strip(), b"vault_usdc"],
+        bump = vault.bumps.vault_usdc)]
+    pub vault_usdc: Box<Account<'info, TokenAccount>>,
+    // Programs and Sysvars
+    pub token_program: Program<'info, Token>,
+}
+
+#[derive(Accounts)]
+pub struct InitEscrowUsdc<'info> {
+    // User Accounts
+    #[account(mut)]
+    pub user_authority: Signer<'info>,
+    #[account(init,
+        token::mint = usdc_mint,
+        token::authority = vault,
+        seeds =  [user_authority.key().as_ref(),
+            vault.vault_name.as_ref().strip(),
+            b"escrow_usdc"],
+        bump,
+        payer = user_authority)]
+    pub escrow_usdc: Box<Account<'info, TokenAccount>>,
+    #[account(seeds = [vault.vault_name.as_ref().strip()],
+        bump = vault.bumps.vault,
+        has_one = usdc_mint)]
```

Comment:
`has_one: enforces the constraint that Increment.counter.authority == Increment.authority.key.`

Do we need to use has_one here? can we use something more readable like 

constraint = vault condition

Reply:
Also I just don't like using has_one - constraints are much more straight forward.

Reply:
Wouldn't the check just then be:
`constraint = vault.usdc_mint == usdc_mint.key()` 
which is just more verbose and achieves the same thing??

Reply:
Yeah more verbose is better than has_one.

Reply:
I'll just leave in for now (imo it makes sense) but point taken on, we can always change later